src/core/src/node.cpp

```diff
@@ -272,9 +272,10 @@ void ov::Node::set_input_is_relevant_to_value(size_t i, bool relevant) {
 }
 
 void ov::Node::set_output_type(size_t i, const element::Type& element_type, const PartialShape& pshape) {
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    get_output_descriptor(i).get_tensor_ptr()->set_tensor_type(element_type, pshape);
-    OPENVINO_SUPPRESS_DEPRECATED_END
+    auto tmp_tensor = std::make_shared<ov::descriptor::Tensor>(element_type,
+                                                               pshape,
+                                                               get_output_descriptor(i).get_tensor_ptr()->get_names());
+    get_output_descriptor(i).set_tensor_ptr(tmp_tensor);
 }
```

Comment:
@ilya-lavrenov  @vurusovs  This change will cause following test failed, any suggestions for this deprecated
![image](https://github.com/openvinotoolkit/openvino/assets/52686861/2197662d-1658-4c67-9c16-681942107ebf)

Reply:
After this line, the parameter 0 layout changed from layout: [N,C] to layout: [...], which caused this error. Continue debugging, any suggestions?
https://github.com/openvinotoolkit/openvino/blob/9bdd36be9c91c952480167dfc23ef2616fd423e4/src/core/src/model.cpp#L855

Reply:
Fixed caused by the rt_info didn't be updated.

Reply:
Updated the rt_info, get another failed, seems like we need to clone all info as following from the org des_tensor to new
https://github.com/openvinotoolkit/openvino/blob/50b2342a541255d96141277b7c3db4d2b932bc27/src/core/include/openvino/core/descriptor/tensor.hpp#L125-L136
this doesn't make sence. I think we do not need to deprecated this API.